components/proxy/src/main/kotlin/com/hotels/styx/services/HealthCheckMonitoringService.kt

```diff
@@ -210,25 +216,16 @@ internal class ObjectDisappearedException : RuntimeException("Object disappeared
 
 
 private fun markObject(db: StyxObjectStore<RoutingObjectRecord>, name: String, newStatus: ObjectHealth) {
-    // The ifPresent is not ideal, but compute() does not allow the computation to return null. So we can't preserve
-    // a state where the object does not exist using compute alone. But even with ifPresent, as we are open to
-    // the object disappearing between the ifPresent and the compute, which would again lead to the compute creating
-    // a new object when we don't want it to. But at least this will happen much less frequently.
-    db.get(name).ifPresent {
-        try {
-            db.compute(name) { previous ->
-                if (previous == null) throw ObjectDisappearedException()
-                val prevTags = previous.tags
-                val newTags = reTag(prevTags, newStatus)
-                if (prevTags != newTags)
-                    it.copy(tags = newTags)
-                else
-                    previous
-            }
-        } catch (e: ObjectDisappearedException) {
-            // Object disappeared between the ifPresent check and the compute, but we don't really mind.
-            // We just want to exit the compute, to avoid re-creating it.
-            // (The ifPresent is not strictly required, but a pre-emptive check is preferred to an exception)
+    db.compute(name) { previous ->
```

Comment:
The same construction is used further up too, on about line 100 or so - this can probably also be reworked in the same way, and the `ObjectDisappearedException` removed.